internal/operands/template-validator/reconcile.go

```diff
@@ -173,15 +172,108 @@ func reconcileDeployment(request *common.Request) (common.ReconcileResult, error
 		Reconcile()
 }
 
-func addPlacementFields(deployment *apps.Deployment, nodePlacement *lifecycleapi.NodePlacement) {
-	if nodePlacement == nil {
+// Merge all Tolerations, Affinity and NodeSelectors from NodePlacement into pod spec
+func injectPlacementMetadata(podSpec *v1.PodSpec, componentConfig *v1beta1.TemplateValidator) {
+	if componentConfig == nil || componentConfig.Placement == nil {
 		return
 	}
+	if podSpec == nil {
```

Comment:
This check can be removed, because the calling function never passes `nil` as `podSpec`. But if you want to keep this check, it can return on `nil`, because the function will be noop.